prosopopee/prosopopee.py

```diff
@@ -229,10 +229,155 @@ def main():
 
     if settings["settings"].get("gm"):
         SETTINGS["gm"].update(settings["settings"]["gm"])
+    return settings
+
+
+def build_gallery(gallery, settings, templates, parent_galleries=False):
+
+    if parent_galleries:
+        gallery_path = os.path.join(parent_galleries, gallery)
+    else:
+        gallery_path = gallery
+
+    gallery_settings = yaml.safe_load(open(os.path.join(os.getcwd(), gallery_path, "settings.yaml"), "r"))
+
+    error(isinstance(gallery_settings, dict), "Your %s should be a dict" % (os.path.join(gallery, "settings.yaml")))
+    error(gallery_settings.get("title"), "You should specify a title in %s" % (os.path.join(gallery, "settings.yaml")))
+
+    gallery_index_template = templates.get_template("gallery-index.html")
+    page_template = templates.get_template("page.html")
+    gallery_cover = {}
+
+    dirs = filter(lambda x: x not in (".", "..") and os.path.isdir(os.path.join(gallery_path, x)) and
```

Comment:
This line would really win at having better variable name or split into several steps with good name or documented.